src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheck.java

```diff
@@ -105,16 +133,54 @@ public void setIgnoreOverridden(boolean ignoreOverridden) {
 
     @Override
     protected boolean mustCheckName(DetailAST ast) {
+        final Scope theScope = calculateScope(ast);
         boolean checkName = true;
         if (ignoreOverridden && isOverriddenMethod(ast)
                 || ast.getParent().getType() == TokenTypes.LITERAL_CATCH
-                || CheckUtils.isReceiverParameter(ast)) {
+                || CheckUtils.isReceiverParameter(ast)
+                || !matchScope(theScope)) {
```

Comment:
please move `calculateScope(ast)` inside expression to speedup a little bit performance, scope calculation is not always required.